### ISSUE TITLE
Scheduling page slots inaccurate for events that are 240 mins long

### ISSUE THREAD

**Issue by wutania** [calcom/cal.com#11744]
### Issue Summary

Scheduling page shows inaccurate number of time slots to book for a 240 min meeting / event. 

### Steps to Reproduce

1. Set your default hours to 5/6 hours each day of the week (eg. 2-8pm or 3-8pm)
2. Set your event to be 240 mins (4 hours) long.
3. Go to scheduling page eg. https://cal.com/betterwater/birthdays
4. Only 1 scheduling slot is available for all days. However, logically, for days with 5 hours of available time, there should be two types of 4-hr scheduling available (eg, for 3-8pm available, there should be two slots: 3-7pm & 4-8pm); and for days with 6 hours of available time, it should be three slots (eg, for 2-8pm, it should be 2-6pm, 3-7pm & 4-8pm).

### Actual Results

- Only 1 scheduling slot is available
### Expected Results

- More than 1 scheduling slot should be available. Eg - 5 hours should have 2 4-hour events that can be booked; 6 hours hours should have 3 4-hour events can be booked.

### Technical details

This was in production: https://cal.com/betterwater/birthdays?date=2023-10-14&month=2023-10

### Evidence

![image](https://github.com/calcom/cal.com/assets/19789656/0bf55c62-4336-4c1f-91b3-5e2306e55bf2)

**Comment by khushalmajoka** [comment]
Hmmm, Nice but these are not slots that you are talking about:

Slots are like: 5 available hours (2pm - 7pm ) | 1 hour meeting time frame
then these these 5 hours will be divided into 5 slots of 1 hour each which will be:

2-3
3-4
4-5
5-6
6-7

and point to note here is that these slots are independent of each other is a slot is booked it can be removed from the available slots but in the other case which you are talking about 5 available hours and 4 hours meeting lets say and if there are 2-6 and 3-7 if out of two one slot is booked other is also booked that means it is of no use and also as the slots are mentioned on cal.com as below:

<img width="656" alt="image" src="https://github.com/calcom/cal.com/assets/56840292/23f2a88b-07aa-4204-aa60-b50118453f85">

it mentions the starting time of slot which is ok in the above case but in your case if we mention

2pm
3pm

then they both are dependent on each other you book either of the slot, that means both are booked.

and one more point is why you would only want the time to be 2pm or 3pm there can be more options if it will be like:

2pm
2:15pm
2:30pm
2:45pm
3:00pm

so basically there are three pointers I guess:

--> available hours
--> meeting hours
--> time gap between each slot and they will be obviously dependent on each other like if a slot is booked the next {meeting hours say 4} hours gone and the slots coming in this time frame too.

**Comment by khushalmajoka** [comment]
okay @wutania , i have found this for you.

Steps to solve your issue:

1. suppose you have an event type of lets say 4 hours

<img width="800" alt="image" src="https://github.com/calcom/cal.com/assets/56840292/da5978a7-027e-43f3-8584-e43923c07d51">

2. also set you availability to 5 hours lets say 2-7

<img width="568" alt="image" src="https://github.com/calcom/cal.com/assets/56840292/30d9e538-67c8-4ca9-ab8c-267c5166a29b">

Now your booking page looks something like this:

<img width="682" alt="image" src="https://github.com/calcom/cal.com/assets/56840292/53b39a14-c9ae-480b-a0b9-884499cddf4e">

3. Now go to the event type you want multiple slot options for and then go to limits tab:

<img width="801" alt="image" src="https://github.com/calcom/cal.com/assets/56840292/d742be5f-4f4e-4b4a-8c3b-84d631dacc13">

4. Here is this time-slot intervals option set it to whichever required (let say I choose 60 mins) instead of the default one and lookout for the booking page.

<img width="663" alt="image" src="https://github.com/calcom/cal.com/assets/56840292/d73bc7a8-896f-45d4-84a4-275aca3bdc62">

The above is what you want i guess.

**Comment by wutania** [comment]
Yooo @khushalmajoka thanks for this example! I think this will solve my problem -> "multiple slot options for and then go to limits tab" 🙏🏼
